From 22a4ce398206380c3b240d7a44d6b2921fb39d46 Mon Sep 17 00:00:00 2001
From: Wolfgang Mauerer <wolfgang.mauerer@siemens.com>
Date: Mon, 21 Oct 2013 15:31:59 +0200
Subject: [PATCH 1/3] Support author name tracking for b4j

Augment the core data structures with a new field to keep
track of the author name. Based on a patch by Deepak Srinathan.

Signed-off-by: Wolfgang Mauerer <wolfgang.mauerer@siemens.com>
---
 src/main/java/b4j/core/DefaultIssue.java    |   13 +++++++++++--
 src/main/java/b4j/core/LongDescription.java |   11 +++++++++++
 2 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/src/main/java/b4j/core/DefaultIssue.java b/src/main/java/b4j/core/DefaultIssue.java
index 7c1b6a7..6e2d16b 100644
--- a/src/main/java/b4j/core/DefaultIssue.java
+++ b/src/main/java/b4j/core/DefaultIssue.java
@@ -1308,7 +1308,8 @@ public class DefaultIssue implements Issue {
 		private Date lastUpdate;
 		private String updateAuthor;
 		private Set<Long> attachments;
-		
+		private String authorName;
+
 		/**
 		 * Default Constructor.
 		 */
@@ -1506,7 +1507,15 @@ public class DefaultIssue implements Issue {
 			return 0;
 		}
 		
-		
+		@Override
+		public String getAuthorName() {
+			return authorName;
+		}
+
+		@Override
+		public void setAuthorName(String authorName) {
+			this.authorName = authorName;
+		}
 	}
 	
 	
diff --git a/src/main/java/b4j/core/LongDescription.java b/src/main/java/b4j/core/LongDescription.java
index c7f3c89..eaa7d6f 100644
--- a/src/main/java/b4j/core/LongDescription.java
+++ b/src/main/java/b4j/core/LongDescription.java
@@ -140,4 +140,15 @@ public interface LongDescription {
 	 */
 	public int getAttachmentCount();
 
+	/**
+	 * Returns the name of the author
+	 * @return author name
+	 */
+	public String getAuthorName();
+
+	/**
+	 * Set the name of the author
+	 * @param author name
+	 */
+	public void setAuthorName(String authorName);
 }
-- 
1.7.10.4

